sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVUtils.scala

```diff
@@ -33,11 +33,12 @@ object CSVUtils {
     // with the one below, `filterCommentAndEmpty` but execution path is different. One of them
     // might have to be removed in the near future if possible.
     import lines.sqlContext.implicits._
-    val nonEmptyLines = lines.filter(length(trim($"value")) > 0)
```

Comment:
@MaxGekk and @cloud-fan, I came up with a better idea to avoid relying on string format in `col`. Can you take a look again? I think this way is safer.